packages/rum-core/src/error-logging/error-logging.js

```diff
@@ -110,10 +110,32 @@ class ErrorLogging {
     }
   }
 
-  registerGlobalEventListener() {
+  registerListeners() {
     window.addEventListener('error', errorEvent =>
       this.logErrorEvent(errorEvent)
     )
+    window.addEventListener('unhandledrejection', promiseRejectionEvent =>
+      this.logPromiseEvent(promiseRejectionEvent)
+    )
+  }
+
+  logPromiseEvent(promiseRejectionEvent) {
+    const prefix = 'Unhandled promise rejection: '
+    const { reason } = promiseRejectionEvent
+
+    if (reason == null) {
+      this.logError(prefix + '<no reason specified>')
+    } else if (typeof reason.message === 'string') {
+      /**
+       * Promise is rejected with an error or error like object
+       */
+      this.logError({
+        message: prefix + reason.message,
+        stack: reason.stack ? reason.stack : null
+      })
+    } else {
+      this.logError(prefix + reason)
```

Comment:
The reason can be an object as well in that case the error message would include `[object Object]`! We should handle the different types of `reason` properly!

Reply:
good point, will move the logic around to make it easier.